modules/data-streams/src/main/java/org/elasticsearch/datastreams/action/DataStreamsStatsTransportAction.java

```diff
@@ -189,6 +191,14 @@ protected DataStreamsStatsAction.DataStreamShardStats readShardResult(StreamInpu
             if (indexAbstraction.getType() == IndexAbstraction.Type.DATA_STREAM) {
                 IndexAbstraction.DataStream dataStream = (IndexAbstraction.DataStream) indexAbstraction;
                 AggregatedStats stats = aggregatedDataStreamsStats.computeIfAbsent(dataStream.getName(), s -> new AggregatedStats());
+                OptionalDouble writeLoadForeCast = clusterState.getMetadata()
+                    .index(indexAbstraction.getWriteIndex())
+                    .getForecastedWriteLoad();
+                if (writeLoadForeCast.isPresent()) {
+                    stats.writeLoadForecast = writeLoadForeCast.getAsDouble();
```

Comment:
Looks better, thanks! However, this is now just the load for one shard of the write index. I think we want to multiply this by the number of shards in that index too.

Actually I can see value in knowing both the per-shard and per-index numbers.

Reply:
Thanks, I added the multiplication with `getNumberOfShards()`.
Do you think I should add the per-shard write load too? Having `writeLoadForecast` and `writeLoadForecastPerShard`?

Reply:
> Do you think I should add the per-shard write load too?

Yes please (let's make both of the names specific: `...PerShard` and `...PerIndex`)

Reply:
> > Do you think I should add the per-shard write load too?
> 
> Yes please (let's make both of the names specific: `...PerShard` and `...PerIndex`)

Done